configure.ac

```diff
@@ -63,7 +63,13 @@ AC_ARG_ENABLE([werror],
 AC_ARG_ENABLE([debug],
     [AS_HELP_STRING([--enable-debug],
                     [Turn on debug output])],
-    [debug=$enableval], [debug=no])
+    [debug=yes], [debug=no])
+
+if test "x${debug}" = "xyes"; then
+  CFLAGS="-Wall -Werror -Wfatal-errors -O0 -g3"
```

Comment:
Fighting with autoconf: I need to find a way to suppress the autoconf default CFLAGS, but only if they are default, otherwise let user defined CFLAGS through.  Unfortunately autoconf doesn't differentiate between default and user supplied CFLAGS, so I need to find an elegant way to do this myself...